src/python/WMCore/MicroService/MSOutput/MSOutput.py

```diff
@@ -248,6 +249,53 @@ def _executeConsumer(self, summary):
             self.logger.exception(msg)
             self.updateReportDict(summary, "error", msg)
 
+    def setDBSStatus(self, workflow):
+        """
+        The function to set the DBS status of outputs as VALID
+        :param workflow: a MSOutputTemplate object workflow
+        :return: the MSOutputTemplate object itself (with the necessary updates in place)
+        """
+        if not isinstance(workflow, MSOutputTemplate):
+            msg = "Unsupported type object '{}' for workflows! ".format(type(workflow))
+            msg += "It needs to be of type: MSOutputTemplate"
+            raise UnsupportedError(msg)
+
+        dbs3Writer = DBS3Writer(url=self.msConfig["dbsReadUrl"],
+                                writeUrl=self.msConfig["dbsWriteUrl"])
+
+        # if anything fail along the way, set it back to "pending"
+        dbsUpdateStatus = "done"
+        for dMap in workflow['OutputMap']:
+
+            if self.msConfig['enableDbsStatusChange']:
+
+                res = dbs3Writer.setDBSStatus(dataset=dMap["Dataset"],
```

Comment:
Note that my comments on the DBS3Writer module will impact this logic.

Another comment, I failed to see where this `self.msConfig['dbsStatus']` gets set in this module. Perhaps I would suggest to define an object variable like:
```
self.dbsOutputStatus = "VALID"
```
and simply refer to it here.

Reply:
I also made it configurable: https://github.com/dmwm/deployment/pull/1044/files#diff-3a773e3fbfb5ff69f5fd806e7a4ab54a7dfaaf1f080e7e96ac1df9e73589c2ffR115 It's also possible to set it within the module. Which one do you prefer?

Reply:
Oh, I missed the deployment changes. Thank for pointing it out.
I think it's fine to keep it in the service configuration file. Nonetheless, I would set a default value to "VALID" in this module, in case the configuration does not have it.

Reply:
Since deployment has the value, I did not set a default value in this module to avoid duplicates. If you think, it's more suitable to keep the configuration in MSOutput module instead of deployment repository, that's also fine, I can change it if you'd like.